pandas/core/generic.py

```diff
@@ -5883,25 +5883,67 @@ def pipe(
         Notes
         -----
         Use ``.pipe`` when chaining together functions that expect
-        Series, DataFrames or GroupBy objects. Instead of writing
+        Series, DataFrames or GroupBy objects.
 
-        >>> func(g(h(df), arg1=a), arg2=b, arg3=c)  # doctest: +SKIP
+        Examples
+        --------
+        Constructing a income DataFrame from a dictionary.
+
+        >>> data = [[8000, 1000], [9500, np.nan], [5000, 2000]]
+        >>> df = pd.DataFrame(data, columns=['Salary', 'Others'])
+        >>> df
+           Salary  Others
+        0    8000  1000.0
+        1    9500     NaN
+        2    5000  2000.0
+
+        Functions that perform tax reductions on an income DataFrame.
+
+        >>> def subtract_federal_tax(df):
+        ...     return df * 0.9
+        >>> def subtract_state_tax(df, rate):
+        ...     return df * (1 - rate)
+        >>> def subtract_national_insurance(df, rate, rate_increase):
+        ...     new_rate = rate + rate_increase
+        ...     return df * (1 - new_rate)
+
+        Instead of writing
+
+        >>> subtract_national_insurance(
+        ...     subtract_state_tax(subtract_federal_tax(df),rate=0.12),
+        ...     rate=0.05,
+        ...     rate_increase=0.02)  # doctest: +SKIP
 
         You can write
 
-        >>> (df.pipe(h)
-        ...    .pipe(g, arg1=a)
-        ...    .pipe(func, arg2=b, arg3=c)
-        ... )  # doctest: +SKIP
+        >>> df.pipe(
+        ...     subtract_federal_tax
+        ... ).pipe(
+        ...     subtract_state_tax, rate=0.12
+        ... ).pipe(subtract_national_insurance, rate=0.05, rate_increase=0.02)
```

Comment:
> Yes, the parentheses are to avoid a flake8 error. But the examples should look as if you were using the interpreter. You wouldn’t be using parentheses in that case.

Does this comment refer to the parenthesis wrapping this whole block? If that's the case, I don't think this is true. The original syntax is much more common and readable than this in my opinion. This is what we've been doing in the docs that I know, it's how code was chained in [the original post about method chaining](https://tomaugspurger.github.io/posts/method-chaining/), it's also the syntax that Matt Harrison uses in the Effective pandas book, and it's what I would write.

To be clear, I'm referring to using
```python
(df.pipe(foo)
   .pipe(bar)
   .pipe(foobar)
)
```
instead of

```python
df.pipe(
    foo
).pipe(
    bar
).pipe(
    foobar
)
```

I'm fine with whatever (even if I personally found the former much more readable), but I wouldn't ask people to change the the latter because the former is not what someone would write, because I think many people would actually write that. :)